config/main.py

```diff
@@ -584,26 +728,59 @@ def reload(filename, yes, load_sysinfo):
     #Stop services before config push
     log_info("'reload' stopping services...")
     _stop_services()
-    config_db = ConfigDBConnector()
-    config_db.connect()
-    client = config_db.get_redis_client(config_db.CONFIG_DB)
-    client.flushdb()
-    if load_sysinfo:
-        command = "{} -H -k {} --write-to-db".format(SONIC_CFGGEN_PATH, cfg_hwsku)
-        run_command(command, display_cmd=True)
 
-    if os.path.isfile(INIT_CFG_FILE):
-        command = "{} -j {} -j {} --write-to-db".format(SONIC_CFGGEN_PATH, INIT_CFG_FILE, filename)
-    else:
-        command = "{} -j {} --write-to-db".format(SONIC_CFGGEN_PATH, filename)
+    """ In Single AISC platforms we have single DB service. In multi-ASIC platforms we have a global DB
+        service running in the host + DB services running in the namespace created per ASIC.
+        In the below logic, we get all namespaces in this platform and add an empty namespace ''
+        denoting the current namespace which we are in ( the linux host )
+    """
+    num_asic = _get_num_asic()
+    for inst in range(-1, num_asic-1):
+        # Get the namespace name, for linux host it is '' empty namespace string.
+        if inst is -1:
+            namespace = ''
+        else:
+            namespace = "{}{}".format(NAMESPACE_PREFIX, inst)
 
-    run_command(command, display_cmd=True)
-    client.set(config_db.INIT_INDICATOR, 1)
+        # Get the config file, based on user input 
+        if len(filename) > inst+1:
+            cfg_file = filename[inst+1]
+        else:
+            if namespace is '':
+                cfg_file = DEFAULT_CONFIG_DB_FILE
+            else:
+                cfg_file = "/etc/sonic/config_db{}.json".format(inst)
 
-    # Migrate DB contents to latest version
-    db_migrator='/usr/bin/db_migrator.py'
-    if os.path.isfile(db_migrator) and os.access(db_migrator, os.X_OK):
-        run_command(db_migrator + ' -o migrate')
+        #Check the file exists before proceeding.
+        if not os.path.isfile(cfg_file):
+            click.echo("The config_db file {} doesn't exist".format(cfg_file))
+            continue
+
+        config_db = ConfigDBConnector(use_unix_socket_path=True, namespace=namespace)
+        config_db.connect()
+        client = config_db.get_redis_client(config_db.CONFIG_DB)
+        client.flushdb()
+        if load_sysinfo:
+            command = "{} -H -k {} -n \"{}\" --write-to-db".format(SONIC_CFGGEN_PATH, cfg_hwsku, namespace)
+            run_command(command, display_cmd=True)
+
+        # For the database service running in linux host we use the file user gives as input
+        # or by default DEFAULT_CONFIG_DB_FILE. In the case of database service running in namespace,
+        # the default config_db<namespaceID>.json format is used.
+
+        if os.path.isfile(INIT_CFG_FILE):
+            command = "{} -j {} -j {} -n \"{}\" --write-to-db".format(SONIC_CFGGEN_PATH, INIT_CFG_FILE, cfg_file, namespace)
```

Comment:
why quote is needed for namespace variable, but not others?

Reply:
This code change was done earlier as per the initial approach for SonicV2Connector/ConfigDBConnector class .. where we had the default namespace as ''. Now that we have changed that to default None, I have updated the code likewise.